worlds/dark_souls_3/__init__.py

```diff
@@ -0,0 +1,281 @@
+        set_rule(
+            self.world.get_entrance("Goto Lothric Castle", self.player),
+            lambda state: state.has("Basin of Vows", self.player))
+        set_rule(
+            self.world.get_location("HWL: Soul of the Dancer", self.player),
+            lambda state: state.has("Basin of Vows", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Irithyll of the boreal", self.player),
+            lambda state: state.has("Small Doll", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Archdragon peak", self.player),
+            lambda state: state.has("Path of the Dragon Gesture", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Profaned capital", self.player),
+            lambda state: state.has("Storm Ruler", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Grand Archives", self.player),
+            lambda state: state.has("Grand Archives Key", self.player))
+        set_rule(
+            self.world.get_entrance("Goto Kiln Of The First Flame", self.player), lambda state:
+            state.has("Cinders of a Lord - Abyss Watcher", self.player) and
+            state.has("Cinders of a Lord - Yhorm the Giant", self.player) and
+            state.has("Cinders of a Lord - Aldrich", self.player) and
+            state.has("Cinders of a Lord - Lothric Prince", self.player))
+
+        self.world.completion_condition[self.player] = lambda state: \
+            state.has("Cinders of a Lord - Abyss Watcher", self.player) and \
+            state.has("Cinders of a Lord - Yhorm the Giant", self.player) and \
+            state.has("Cinders of a Lord - Aldrich", self.player) and \
+            state.has("Cinders of a Lord - Lothric Prince", self.player)
+
+    def generate_basic(self):
+        # Depending on the specified option, add the Basin of Vows to a specific location or to the item pool
+        item = self.create_item("Basin of Vows")
+        if self.world.late_basin_of_vows[self.player]:
+            self.world.get_location("IBV: Soul of Pontiff Sulyvahn", self.player).place_locked_item(item)
+        else:
+            self.world.itempool += [item]
+
+        # Fill item pool with additional items
+        item_pool_len = self.item_name_to_id.__len__()
+        total_required_locations = self.location_name_to_id.__len__()
+        for i in range(item_pool_len, total_required_locations):
+            self.world.itempool += [self.create_item("Soul of an Intrepid Hero")]
+
+    def generate_output(self, output_directory: str):
+        # Depending on the specified option, modify items hexadecimal value to add an upgrade level
+        item_dictionary = item_dictionary_table.copy()
+        if self.world.randomize_weapons_level[self.player]:
+            # Randomize some weapons upgrades
+            for name in weapons_upgrade_5_table.keys():
+                if randint(0, 100) < 33:
```

Comment:
`self.world.random` object is seeded per generation and is deterministic. please use it instead

Reply:
Fixed